### ISSUE TITLE
RFE: Printing with syntax highlighting

### ISSUE THREAD

**Issue by mickep76** [fileability/chocolat-public#1757]
Currently printing Code doesn't include any formatting, it would be nice if it could format the code when printing like "enscript" does.

**Comment by atg** [comment]
Yeah this is another thing that depends on rewriting the syntax highlighting system (#1762). Currently it's not modular _at all_, and so coloring it before a print is pretty much impossible.

As it stands, Chocolat does not color the whole document at once, it only colors the visible area. Obviously this is no good for printing.
